src/v/v8_engine/script.h

```diff
@@ -84,10 +92,29 @@ class script {
         void operator()(v8::Isolate* isolate) const { isolate->Dispose(); }
     };
 
+    // Stop v8 execution if it takes too much time
+    void stop_execution();
+    // Cancel stop execution for reusing isolate
+    void continue_execution();
+
     std::unique_ptr<v8::Isolate, isolate_deleter> _isolate;
 
     v8::Global<v8::Context> _context;
     v8::Global<v8::Function> _function;
+
+    executor& _executor; // TODO: maybe use unique_pointer without destructor
+                         // for move v8_script in container with scripts
+
+    // Watchdog for script timelimit
+    seastar::timer<seastar::lowres_clock> _watchdog;
```

Comment:
Maybe the time limit should be an option to `executor::submit` and that class can manage time limits.

Reply:
I think not, Because when people run smth like `rpk wasm create script` they need to give timeout for script

How can you manage timeout for each `v8_engine::script.run`

Reply:
it would work the same way, just the executor would be responsible for arming the timer and managing it

Reply:
Oh, I understand
Hmmm, interesting

Reply:
I wrote first version for this

Reply:
Done